src/status_im/contexts/wallet/events.cljs

```diff
@@ -42,8 +43,18 @@
          [:dispatch [:wallet/show-account-created-toast address]]]}))
 
 (rf/reg-event-fx :wallet/switch-current-viewing-account
- (fn [{:keys [db]} [address]]
-   {:db (assoc-in db [:wallet :current-viewing-account-address] address)}))
+ (fn [{:keys [db]} [{:keys [address network-details update-balance?]}]]
+   (let [current-token (-> db :wallet :ui :send :token)
```

Comment:
hmm I would like to push back against this change. We're taking a very simple and effective method and coupling it with some send related logic. 
Can we handle this some other way? - maybe just a separate event being fired for this particular action? 🤔 

cc @smohamedjavid

Reply:
@J-Son89  @briansztamfater 

According to the comment in this event:
```
;; When switching accounts, if there is a selected token we need to update
;; it to reflect balances of the new account
```

Seems more as a taks for a subscription, the subscription that extracts the token from
 `[:wallet :ui :send :token]`
could also receive the current-viewing-account and add the extra data needed.

wdyt?

Reply:
I think it makes sense, will try to refactor this and let you know!

Reply:
I removed this logic, send the updated token from subscription to `:wallet/get-suggested-routes` event